ME branches and reweighting re-written

usarica opened the pull request:
- test area scripts need checking, may not compile/use incorrect MEs
- csv files not edited at the moment.

hroskes committed:
better reweighting

hroskes committed:
fix in case probability=0
If this is true for all events (e.g. not enough leptons) you shouldn't be reweighting so can ignore the weights.
If it happens for 1 event at some crazy phase space point you want to exclude it anyway.

hroskes committed:
oops

hroskes committed:
Merge branch 'miniAOD_80X' of https://github.com/CJLST/ZZAnalysis into miniAOD_80X

hroskes committed:
No need to specify spin separately

hroskes committed:
Merge branch 'miniAOD_80X' of https://github.com/CJLST/ZZAnalysis into miniAOD_80X_betterreweighting

hroskes committed:
Do reweighting calculation using the LHE event

hroskes committed:
reweighting for VBF+decay

hroskes committed:
Fix electron scale checkout tag, gitignore

hroskes committed:
can't just add the candidate

hroskes committed:
Need to set couplings again before calculating second ME

hroskes committed:
Fixes

hroskes committed:
rearrange a bit

hroskes committed:
Implement reweighting cutoffs

hroskes committed:
anomalous production probabilities

hroskes committed:
Merge branch 'miniAOD_80X' of https://github.com/CJLST/ZZAnalysis into miniAOD_80X_betterreweighting

hroskes committed:
Fix TLEs, RSEs, CR

hroskes committed:
VVMode = 1 by default
(we do mostly ZZ after all)

hroskes committed:
fix stupid bug, and use simpler setCurrentCandidate
(checkout_80X.csh needs to be updated once mela has a tag)

hroskes committed:
VH anomalous couplings MEs, reweighting, and samples

hroskes committed:
Merge branch 'miniAOD_80X' of https://github.com/CJLST/ZZAnalysis into miniAOD_80X_betterreweighting

Conflicts:
	checkout_80X.csh

hroskes committed:
script to calculate the reweighting cutoffs that should be used

hroskes committed:
script to replace all reweighting cutoffs in a file, and use it

hroskes committed:
oops

hroskes committed:
cutoffs for WH
remove root option from the script since it's not that much faster

Commit:
Start to generalize MC reweigthing for all couplings defined in MELA

Commit:
Missed a couple lines

hroskes committed:
Merge branch 'miniAOD_80X' of https://github.com/CJLST/ZZAnalysis into miniAOD_80X_betterreweighting

hroskes committed:
Merge branch 'miniAOD_80X_newReWgt' of github.com:usarica/ZZAnalysis into miniAOD_80X_betterreweighting

Conflicts:
	AnalysisStep/interface/reweighting.h
	AnalysisStep/python/couplings.py
	AnalysisStep/src/reweighting.cc
	AnalysisStep/test/Ntuplizers/HZZ4lNtupleMaker.cc
	AnalysisStep/test/analyzer.py

usarica committed:
Merge pull request #1 from hroskes/miniAOD_80X_betterreweighting

merge reweighting updates and prod MEs

usarica committed:
Somehow compiles fine, not sure why

usarica committed:
ExtendedBranch class to generalize how branches are booked

usarica committed:
Merge miniAOD_80X_newReWgt

usarica committed:
Add ME computation class

usarica committed:
MELA computation class complete (getBranchType issue aside)

usarica committed:
Proper type checking

usarica committed:
Implement methods to identify ME specifics given pyFragments syntax

usarica committed:
Add computation class

usarica committed:
MELA routines to ZZCandidateFiller

usarica committed:
ZZCand option to branch ME

hroskes committed:
HJJ and ttH anomalous samples and production matrix elements for samples that need it

hroskes committed:
Merge branch 'miniAOD_80X' of github.com:CJLST/ZZAnalysis into miniAOD_80X_betterreweighting

Conflicts:
	AnalysisStep/test/prod/samples_2016.csv

usarica committed:
Merge pull request #2 from hroskes/miniAOD_80X_betterreweighting

samples and update

usarica committed:
More improvements

usarica committed:
Merge origin

usarica committed:
Implement copy options

usarica committed:
Extended couplings for ggH production

usarica committed:
ME branches now in ZZCandidateFiller

usarica committed:
Implement LHE and Reco MELA branches

usarica committed:
Revert changes to classes

usarica committed:
Some mods

usarica committed:
Make it compile

usarica committed:
Almost there: No working 0J ME, booking pConst instead of pAux, inconclusive leptonic VH

usarica committed:
Fixed pAux

usarica committed:
Fix decay MEs=0

usarica committed:
All reco validated

usarica committed:
Found inconsistent ttH, else fine

usarica committed:
Fix ttH (issue was maximizationcache), speedup

usarica committed:
Add and check bestDJJ

usarica committed:
Wipe out old MEs

usarica committed:
LHE weights valid

usarica committed:
Ready for a review

namapane commented:
I had a quick look at the diffs and it looks OK, but hadn't had the time to test it. Did you try to run the usual synchronization? I suppose we should expect all discriminant to change slightly given the changes in MELA, right?

namapane commented:
Hi, I started testing this but I see that the checkout recipe does not include the new MELA version, am I right?

namapane commented:
@usarica , I think you should replace L50 in checkout_80X.csh with:
(cd ZZMatrixElement ; git checkout -b from-v2.0.1.b1 v2.0.1.b1)

and in addition, you should modify
ZZMatrixElement/MELA/data/slc6_amd64_gcc530/download.url
to pick libmcfm_702.so 
Can you please do these modifications?

Thanks
N. 
(@covarell : FYI)

usarica committed:
Wipe MEs from csv files first

usarica committed:
Add reco MEs back

usarica committed:
Add some other pyFragments (TBC)

usarica committed:
Updated Mela

usarica committed:
Fragment builder

usarica committed:
Auto MELA fragment maker for ggH POWHEG

usarica committed:
VBF, VH fragments

usarica committed:
Comment out many unneeded MEs

usarica committed:
Main CSV file ready

usarica committed:
Minor fixes to fragments

usarica committed:
Rename files

usarica committed:
Update AC csv file for some fragments

usarica committed:
ttH and HJJ fragments, cluster computation slightly more compact

usarica committed:
Fix HJJ and ttH files

usarica committed:
All changes are done, I hope...

usarica committed:
Why did I have the impression that everything was done anyway?

usarica committed:
Resolve conflicts

usarica committed:
Why missing p_? Not sure.

usarica committed:
Per discussion with Raquel, added prod-decay MCFM MEs for VBF and QCD

hroskes committed:
script to test the syntax of the python fragments

hroskes committed:
fix variable name

hroskes committed:
fix git checkout

hroskes committed:
fix )

usarica committed:
Merge pull request #3 from hroskes/miniAOD_80X_newReWgt

Fix variable name

hroskes committed:
remove old variable that doesn't do anything

hroskes committed:
copy default recoProbabilities to TLE and RSE

hroskes committed:
fix typo in pyFragment name

usarica committed:
Merge pull request #4 from hroskes/miniAOD_80X_newReWgt

a few more small bugs

hroskes committed:
allow running locally (with saving the exit status etc.)

hroskes committed:
add probabilities to the other ZZCandidates (and not just the trees)

hroskes committed:
only if it exists (ZZCandlooseEle and its friends sometimes don't)

hroskes committed:
Merge branch 'miniAOD_80X' of https://github.com/CJLST/ZZAnalysis into miniAOD_80X_newReWgt

Conflicts:
	AnalysisStep/test/prod/samples_2016.csv

hroskes committed:
add probabilities to 2017 processed data

hroskes committed:
remove nonsense in comment

hroskes committed:
add probabilities to 2017 MC

usarica committed:
Merge pull request #5 from hroskes/miniAOD_80X_newReWgt

Fix another bug, and merge

hroskes committed:
don't add lheProbabilities to candidates

usarica committed:
Merge pull request #6 from hroskes/miniAOD_80X_newReWgt

don't add lheProbabilities to candidates

hroskes committed:
fix CPS reweighting

hroskes committed:
use isProp option instead

usarica committed:
Merge pull request #7 from hroskes/miniAOD_80X_newReWgt

fix CPS reweighting

hroskes committed:
close quotes

usarica committed:
MELA update 701b3

usarica committed:
Merge pull request #8 from hroskes/miniAOD_80X_newReWgt

close quotes

hroskes committed:
Merge branch 'miniAOD_80X' of https://github.com/CJLST/ZZAnalysis into miniAOD_80X_newReWgt

hroskes committed:
separate failedTree for for events with no candidate

hroskes committed:
turn on failedTreeLevel for samples with reweighting

hroskes committed:
change name, fix

hroskes committed:
That is very confusing... int32 does not implement inequality operators, only == and !=

hroskes committed:
don't need this line anymore

hroskes committed:
Merge branch 'miniAOD_80X' of https://github.com/CJLST/ZZAnalysis into miniAOD_80X_newReWgt

Conflicts:
	AnalysisStep/test/prod/samples_2016ICHEP_Data.csv
	AnalysisStep/test/prod/samples_2017mc.csv

hroskes committed:
keep event in candTree_failed even if it fails trigger or skim

hroskes committed:
fix ghz --> ghv for WH (consistently)

hroskes committed:
MCFM signal and BSI reweighting

hroskes committed:
change WH probabilities to ghw

hroskes committed:
fix HadVH --> Had_VH

hroskes committed:
bookkeep MELABranches a bit differently

hroskes committed:
phantom does not use CPS

hroskes committed:
allow copying when the folder is a symbolic link

(long story)

(when there's a bug that affects a few folders, the easiest way for me
is to make symbolic links to them in another folder and then do resubmit.csh
there.  cp -rf $LS_SUBCWD . copies the link, cp -rf $LS_SUBCWD/ . copies
the target)

hroskes committed:
fix VVDECAYMODE for ttH

(including the POWHEG and systematics samples,
we don't reweight now but we could potentially)

namapane committed:
Fix sync script for missing probabilities

hroskes committed:
Merge branch 'miniAOD_80X_newReWgt' of github.com:usarica/ZZAnalysis into miniAOD_80X_newReWgt

namapane committed:
Minor syntax fix in sync script

namapane committed:
Fix variable names in synchonizer.py

hroskes committed:
Merge branch 'miniAOD_80X_newReWgt' of github.com:usarica/ZZAnalysis into miniAOD_80X_newReWgt

hroskes committed:
lepton flavors

hroskes committed:
disable ttH production reweighting

hroskes committed:
turn failedTreeLevel into an enum

with some acrobatics to get it to be readable in C++ and python

hroskes committed:
remove buggy 4tau events from MCFM

hroskes committed:
Merge branch 'miniAOD_80X' of https://github.com/CJLST/ZZAnalysis into miniAOD_80X_newReWgt

hroskes committed:
fix name

hroskes committed:
Merge branch 'miniAOD_80X' of https://github.com/CJLST/ZZAnalysis into miniAOD_80X_newReWgt

hroskes committed:
turn off gamma* and imaginary probabilities for phantom

hroskes committed:
Merge branch 'miniAOD_80X' of https://github.com/CJLST/ZZAnalysis into miniAOD_80X_newReWgt

Conflicts:
	AnalysisStep/test/prod/samples_2016_MC.csv

hroskes committed:
Merge branch 'miniAOD_80X' of https://github.com/CJLST/ZZAnalysis into miniAOD_80X_newReWgt

hroskes committed:
update MELA version

hroskes committed:
turn back on SM BSI imaginary

usarica committed:
Merge pull request #9 from hroskes/miniAOD_80X_newReWgt

put failed events into a separate tree

usarica committed:
Delete calcreweightingcutoffs.py

hroskes committed:
same bug happens in 2e2tau 2mu2tau

usarica committed:
Merge pull request #10 from hroskes/miniAOD_80X_newReWgt

same bug happens in 2e2tau 2mu2tau

hroskes committed:
use the same naming convention everywhere for WH ME

hroskes committed:
don't use ROOT!

usarica committed:
Merge pull request #11 from hroskes/miniAOD_80X_newReWgt

use the same naming convention everywhere for WH ME

hroskes committed:
most ggH samples are available

hroskes committed:
pythia systematics samples are on DAS too

hroskes committed:
turn on phantom bkg samples that are ready

hroskes committed:
change c constants for VH

hroskes committed:
remove c constant for Dbkg

hroskes committed:
Merge branch 'miniAOD_80X' of https://github.com/CJLST/ZZAnalysis into miniAOD_80X_newReWgt

Conflicts:
	AnalysisStep/test/Ntuplizers/HZZ4lNtupleMaker.cc

hroskes committed:
only enable failedTreeLevel for ZZTree

usarica committed:
Merge
